Model/Checkout/GetQuote.php

```diff
@@ -288,7 +288,7 @@ public function createQuote(array $orderItems): Quote
         $quote->setStore($this->storeManager->getStore());
         $quote->setCurrency();
 
-        if ($customer = $this->getCustomer()) {
+        if ($customer = $this->getSessionCustomer()) {
             $quote->assignCustomer($customer);
         } else {
```

Comment:
Only setCustomerIsGuest if the quote doesn't have customer id. 
```suggestion
        } else if (!quote->getCustomerId()) {
```